### ISSUE TITLE
flatline rule

### ISSUE THREAD

**Issue by xiaoxinba** [Yelp/elastalert#1279]
my scene is:
     In the period of one minute, if there is no event log,then alert;
 my rule:
   `type: flatline
   threshold: 1
   realert:
       minutes: 1
   query_key: from
   query_key: from`
  run_query:1
  buffer_time:1

but some times ,there are logs but elastalert send alert,why? delay? or ...
please help!!

**Comment by Trustys** [comment]
What is your run_every settings in config.yaml ?

**Comment by Paraomao** [comment]
Hello,

I have a very similar issue:

General Elastalert config:
run_every:
  minutes: 5

buffer_time:
  hours: 8

--------------
The rule:
name: 'My test rule'

type: flatline

use_strftime_index: true
index: logstash-test-%Y.%m.%d

threshold: 1

timeframe:
  hours: 2

doc_type: test

use_terms_query: true
query_key: sys.id.keyword
terms_size: 500

filter:
- query:
    query_string:
      query: '* NOT sys.mtype.keyword:"test"'

raw_count_keys: False
summary_table_fields:
  - sys.id.keyword

aggregation:
  schedule: '00 08 * * 1-5'

alert:
[...]
-----------

In ElasticSearch I have matching events from the same sys.id.keyword every 10 minutes without a significant delay (seconds), so I do not expect any alert from Elastalert, instead  Elastalert trigger alerts systematically for all the sys.id.keyword.

It seems that the timeframe is ingored.

BR

**Comment by Qmando** [comment]
@xiaoxinba 
"sometimes there are logs but elastalert sends an alert"
Do they have the same ``from`` value? If you set a query_key, an alert will happen if any unique value of that ``query_key`` stops for 1 minute.

**Comment by Qmando** [comment]
@Paraomao 
the timeframe is NOT ignored.
Same response as to the original poster, it will alert if any single value of sys.id.keyword appears and then stops for 2 hours. Maybe there is the possibility that the 500 size is not big enough and it's missing some terms. Does it always say there are less than 500 buckets for each query?

**Comment by xiaoxinba** [comment]
@Qmando 
Yes, I know that if set a query_key, an alert will happen if any unique value of that query_key stops for 1 minute.
But , for the from(such as host1),even there are some log for host1, but alert for host1 ,Why?

**Comment by Paraomao** [comment]
@Qmando 
I'm using the rule to monitor the heartbeat from 25 servers with different sys.id.keyword, that heartbeat is sent by each server every 5 minutes.

The terms_size should be at least:
25 (servers) * 24 (events for server in two hours) = 600 (so my terms_size is too low)

This is correct?
Thank you
